core/src/main/scala/kafka/api/LeaderAndIsr.scala

```diff
@@ -43,6 +44,7 @@ case class LeaderAndIsr(leader: Int,
   def newEpochAndZkVersion = newLeaderAndIsr(leader, isr)
 
   override def toString: String = {
-    Json.encode(Map("leader" -> leader, "leader_epoch" -> leaderEpoch, "isr" -> isr))
+    val isrList = if (isr.size !=0) isr else null
+    return "LeaderAndIsr(leader=" + leader + ", leader_epoch=" + leaderEpoch + ", isr=" + isrList + ", zk_version=" + zkVersion + ")"
```

Comment:
We can use string interpolation here. The field name should be used (i.e. camel-case). `return` should not be used,

Reply:
Ok.